api/v1/cluster_webhook.go

```diff
@@ -895,6 +895,10 @@ func (r *Cluster) validateConfiguration() field.ErrorList {
 		}
 	}
 
+	// verify the postgres setting min_wal_size < max_wal_size < volume size
+	result = append(result, validateWalSizeConfiguration(
+		r.Spec.PostgresConfiguration, r.Spec.WalStorage.GetSizeOrNil())...)
+
```

Comment:
if user did not specific walStorage, do we need to consider using the size of storage like

	if r.ShouldCreateWalArchiveVolume() {
		walVolumeSize = r.Spec.WalStorage.GetSizeOrNil()
	} else {
		walVolumeSize = r.Spec.StorageConfiguration.GetSizeOrNil()
	}

Reply:
For the same reason as the previous comment, that would give the user a false sense of correctness. The storage area is meant to keep the data, not the WAL files. We can only validate the user input if a walStorage section is present.